temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -712,20 +711,28 @@ private void processScheduleNexusOperation(
       long workflowTaskCompletedId) {
     attr = validateScheduleNexusOperation(attr);
     Endpoint endpoint = nexusEndpointStore.getEndpointByName(attr.getEndpoint());
-    StateMachine<StateMachines.NexusOperationData> operationStateMachine =
-        newNexusOperation(endpoint);
+    StateMachine<StateMachines.NexusOperationData> operation = newNexusOperation(endpoint);
     long scheduleEventId = ctx.getNextEventId();
-    nexusOperations.put(scheduleEventId, operationStateMachine);
+    nexusOperations.put(scheduleEventId, operation);
 
-    operationStateMachine.action(Action.INITIATE, ctx, attr, workflowTaskCompletedId);
+    operation.action(Action.INITIATE, ctx, attr, workflowTaskCompletedId);
+    // Use TIMEOUT_TYPE_SCHEDULE_TO_START for timing out an individual start request
+    ctx.addTimer(
+        ProtobufTimeUtils.toJavaDuration(operation.getData().taskRetryPolicy.getInitialInterval()),
+        () ->
+            timeoutNexusOperation(
+                scheduleEventId,
+                TimeoutType.TIMEOUT_TYPE_SCHEDULE_TO_START,
+                operation.getData().getAttempt()),
+        "ScheduleNexusOperation ScheduleToStart");
     ctx.addTimer(
         ProtobufTimeUtils.toJavaDuration(
-            operationStateMachine.getData().scheduledEvent.getScheduleToCloseTimeout()),
+            operation.getData().scheduledEvent.getScheduleToCloseTimeout()),
```

Comment:
This isn't required, make sure you don't set a timer unless this timeout is set.